File: C/hl.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    int guess1;
    int guess2;
    time_t t;
    char highlow;
    char yn;

    while(1){
        int score = 0;
        while(1){
            // chooses a new random number everytime the program runs
            srand((unsigned) time(&t));

            guess1 = rand() % 10;
            printf("------------------------------------------------------------------------\n");
            printf("The computer has picked the number %d\n", guess1);

            guess2 = rand() % 10;

            printf("Is the computer's second number going to be higher (h) or lower (l) than the first? ");
            // there MUST be a space before %c otherwise the program will loop back to the start from here
            scanf(" %c", &highlow); // & stores the input in the variable

            if (guess1 > guess2 && highlow == 'l'){
                printf("Correct! The second number is %d", guess2);
                printf("\n");
                score = score + 100;
                printf("Your score = %d", score);
                printf("\n");
            } else if (guess1 < guess2 && highlow == 'h'){
                printf("Correct! The second number is %d", guess2);
                printf("\n");
                score = score + 100;
                printf("Your score = %d", score);
                printf("\n");
            } else if (guess1 > guess2 && highlow == 'h'){
                printf("Wrong! The second number is %d", guess2);
                printf("\n");
                printf("You finished the game with %d points", score);
                printf("\n");
                break;
            } else if (guess1 < guess2 && highlow == 'l'){
                printf("Wrong! The second number is %d", guess2);
                printf("\n");
                printf("You finished the game with %d points", score);
                printf("\n");
                break;
            } else if (guess1 == guess2) {
                printf("They are the same! The second number is %d", guess2);
                printf("\n");
            }
        }
        printf("Play again? (y/n)");
        scanf(" %c", &yn);
        if (yn == 'n'){
            break;
        }
    }
    return 0;
}

File: Go/hl.go
package main

import (
	"fmt"
	"math/rand/v2"
)

func main() {
	var highlow string
	var yn string

	for {
		var score = 0
		fmt.Println(">>0")

		for {
			fmt.Println()
			fmt.Println("-----------------------------------")
			guess1 := rand.IntN(11)
			fmt.Println("The program has picked the number", guess1)
			fmt.Println("Is the next number going to be higher (h) or lower (l) than", guess1, "?")
			fmt.Scan(&highlow)
			guess2 := rand.IntN(11)

			if highlow == "h" || highlow == "l" {
				if highlow == "h" && guess2 > guess1 {
					fmt.Println("Correct!", guess2, "is larger than", guess1)
					score = score + 100
					fmt.Println("+100 points!")
					fmt.Println("Your score =", score)
				} else if highlow == "h" && guess2 < guess1 {
					fmt.Println("Wrong!", guess2, "is smaller than", guess1)
					fmt.Println("You finished the game with a score of", score)
					break
				} else if highlow == "l" && guess1 > guess2 {
					fmt.Println("Correct!", guess2, "is smaller than", guess1)
					score = score + 100
					fmt.Println("+100 points!")
					fmt.Println("Your score =", score)
				} else if highlow == "l" && guess1 < guess2 {
					fmt.Println("Wrong!", guess2, "is larger than", guess1)
					fmt.Println("You finished the game with a score of", score)
					break
				} else {
					fmt.Println(guess1, "and", guess2, "are the same!")
					fmt.Println("+0 points")
					fmt.Println("Your score =", score)
				}
			} else {
				fmt.Println("Invalid input. Must be either higher (h) or lower (l)")
			}
		}

		fmt.Println()
		fmt.Println("Would you like to play again? (y/n)")
		fmt.Scan(&yn)
		if yn == "n" {
			break
		}
		score = 0
		fmt.Println(">>0")
	}

}
